### ISSUE TITLE
Registers help menu not updating on * press

### ISSUE THREAD

**Issue by Nemoos-0** [helix-editor/helix#7769]
### Summary

When pressing `*` I expect the `Registers` help menu to update with the current selection in the `/` register.

The search actually works fine and if instead of `*` I use `"/y` the `Registers` help menu gets updated correctly.

### Reproduction Steps

 

I tried this:

1. `helix --tutor`
2. `/Welcome<enter>`
3. `ww*`
4. `"`

I expected this to happen:
The `Registers` help menu should show `/ to`

Instead, this happened:
It shows `/ Welcome`

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-07-28T12:09:34.237 helix_view::clipboard [DEBUG] Using xclip to interact with the system and selection (primary) clipboard
2023-07-28T12:09:34.240 helix_vcs [INFO] Error {
    context: "failed to open git repo",
    source: Discover(
        NoGitRepository {
            path: "/usr/lib/helix/runtime",
        },
    ),
}
2023-07-28T12:09:34.240 helix_vcs [INFO] failed to open diff base for /usr/lib/helix/runtime/tutor
2023-07-28T12:09:34.240 helix_vcs [INFO] Error {
    context: "failed to open git repo",
    source: Discover(
        NoGitRepository {
            path: "/usr/lib/helix/runtime",
        },
    ),
}
2023-07-28T12:09:34.240 helix_vcs [INFO] failed to obtain current head name for /usr/lib/helix/runtime/tutor
2023-07-28T12:09:34.254 helix_tui::backend::crossterm [DEBUG] The keyboard enhancement protocol is not supported in this terminal (checked in 13.466653ms)
2023-07-28T12:09:34.255 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:34.639 helix_term::application [DEBUG] received editor event: IdleTimer
2023-07-28T12:09:35.948 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:36.349 helix_term::application [DEBUG] received editor event: IdleTimer
2023-07-28T12:09:36.543 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:36.974 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:37.404 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:38.015 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:38.315 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:38.604 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:38.759 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:39.624 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:40.836 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:41.156 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:41.556 helix_term::application [DEBUG] received editor event: IdleTimer
2023-07-28T12:09:42.430 helix_view::editor [DEBUG] editor status: register '/' set to 'to '
2023-07-28T12:09:42.431 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:42.830 helix_term::application [DEBUG] received editor event: IdleTimer
2023-07-28T12:09:43.581 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:43.982 helix_term::application [DEBUG] received editor event: IdleTimer
2023-07-28T12:09:46.021 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:46.421 helix_term::application [DEBUG] received editor event: IdleTimer
2023-07-28T12:09:46.666 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:47.068 helix_term::application [DEBUG] received editor event: IdleTimer
2023-07-28T12:09:47.637 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:47.977 helix_term::commands::typed [DEBUG] quitting...
2023-07-28T12:09:47.977 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:48.037 helix_term::application [DEBUG] received editor event: IdleTimer
2023-07-28T12:09:48.556 helix_term::commands::typed [DEBUG] quitting...
2023-07-28T12:09:48.557 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-07-28T12:09:48.557 helix_term::job [DEBUG] waiting on jobs...
2023-07-28T12:09:48.557 helix_term::job [DEBUG] waiting on jobs...
```

</details>

### Platform

Linux

### Terminal Emulator

wezterm 20230408-112425-69ae8472

### Helix Version

helix 23.05

**Comment by the-mikedavis** [comment]
Duplicate of https://github.com/helix-editor/helix/issues/5241
